pkg/plugins/manager.go

```diff
@@ -0,0 +1,128 @@
+package pluginmanager
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+
+	"github.com/massalabs/thyra/pkg/config"
+)
+
+type PluginManifest struct {
+	Name    string
+	Bin     string
+	Version string
+}
+
+type PluginItem struct {
+	Manifest PluginManifest
+	Path     string
+	Port     int
+}
+
+type PluginManager struct {
+	HostPort    int
+	HostTLSPort int
+	plugins     []PluginItem
+}
+
+func (manager *PluginManager) List() []PluginItem {
+	return manager.plugins
+}
+
+func DetectPlugin(path string) (*PluginManifest, error) {
+	manifestFileName := "manifest.json"
+
+	manifestPath := filepath.Join(path, manifestFileName)
+
+	if _, err := os.Stat(manifestPath); err != nil {
+		return nil, errors.New("unable to find manifest.json in " + path)
+	}
+
+	jsonFile, err := os.Open(manifestPath)
+	if err != nil {
+		return nil, errors.New("unable to open manifest.json in " + path)
+	}
+
+	var manifest PluginManifest
+
+	jsonBytes, _ := io.ReadAll(jsonFile)
+
+	err = json.Unmarshal(jsonBytes, &manifest)
+
+	if err != nil {
+		return nil, errors.New("unable to decode json manifest in " + path)
+	}
+
+	return &manifest, nil
+}
+
+func StartPlugin(plugin PluginItem) {
+	log.Println("Starting plugin '" + plugin.Manifest.Name + "' on port " + strconv.Itoa(plugin.Port))
+
+	cmd := exec.Command(filepath.Join(plugin.Path, plugin.Manifest.Bin),
+		"--port", strconv.Itoa(plugin.Port), "--path", plugin.Path) // #nosec G204
```

Comment:
It's not an error : gosec is telling you that you are using the command function, see [here](https://pkg.go.dev/github.com/securego/gosec/v2#readme-available-rules) for details.

You should add a comment explaining that and explaining why your code is safe. Is it safe ?

Reply:
Imho Its not safe att all. but its the solution we chose for now.. and it will be possible to add security layers later. like a comparing a checksum of the binary for example...
but ok i can add a comment

Reply:
but now i am using a package to handle the command so i dont have to disable the litter rule anymore